server/src/test/java/org/apache/druid/metadata/IndexerSQLMetadataStorageCoordinatorTest.java

```diff
@@ -1068,6 +1068,7 @@ public void testMultiIntervalUsedList() throws IOException
         coordinator.retrieveUsedSegmentsForIntervals(
             defaultSegment.getDataSource(),
             ImmutableList.of(defaultSegment.getInterval()),
+            null,
```

Comment:
Should we add tests here that test with version specified, or does something else exercise that code path?

Reply:
Thanks. Added a few unit tests in this test as well. `RetrieveSegmentsActionsTest` has coverage of this functionality too.